### PR TITLE
BL-727/Hided features that are not working

### PR DESCRIPTION
**Pull request by khaleel737**
## Description

Hidden Non-functional and empty components of Admin, Mentor, Mentee users and made a new account with role_id = 5 which is for development purposes that displays all the components of Admin, Mentor, Mentee. 

Created a new Dashboard Component for future work.

This is the current Application status in regards to every component built, shows the fully functional, functional with bugs and non-functional features built, here is also screenshots of the current dashboards we have, Ive made a Dev account that shows all the features into a single account for Developers, makes it easier to work and reflect your work on all the dashboards of Admin, Mentor, Mentee users.

File of functional and non-funtional features:
https://docs.google.com/spreadsheets/d/1JYt8FXOwxH_hV8QIISRCYkJ_qZo6pPY9XCYj8QY_iGU/edit#gid=0

Fixes # (issue)
Memo redirected to mynotes now fixed to redirect to memos.
Refactored role side bar display with strict equality for each role. 
Changed Dashboard.js to TicketsDashboard.js and rerouted the page to dashboard side link, made a new Dashboard component that is set to the main path (/).

## Loom Video

https://www.loom.com/share/aaab38db2bcd4b0194609a0b14dde7d5

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by stevenhastings — DISMISSED** [review]
Job well done! I didn't see a thing.  We should all come together to work on the ticket, BL-717-kick-auth0-to-the-curb.  Great job Khaleel, as always.

**Review by jtorres03 — DISMISSED** [review]
Very nice! From what I can tell the changes you made make it easier to navigate for those contributing to devs but also removes unnecessary features for mentees/mentors. Well done and well compartmentalized!

**Review by ezra-casas — DISMISSED** [review]
This is great stuff.

**Comment by jack-semder** [comment]
Nice dashboard implantation, that's a huge feature. Also good work on ridding of hidden/irrelevant components!

**Review by jack-semder — DISMISSED** [review]
Nice dashboard implementation, that's a huge feature. Also good work on ridding of hidden/irrelevant components!

**Review by Charles-Merrill-Johnson — DISMISSED** [review]
Great job cleaning this up. I like the use of the ternary operator. And implementing the links

**Comment by andrewgomez1992** [comment]
Looks great will help us with navigating for sure!

**Review by farhaannishtar — DISMISSED** [review]
Code is clean and readable. I don't see any issues. Great work.

File: src/components/common/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -38,8 +39,21 @@ const Sidebar = ({ children, userProfile }) => {
...
+   */
+
   const isUserAdmin = useMemo(() => role_id <= 2, [role_id]);
+  const isUserMentor = useMemo(() => role_id === 3, [role_id]);
+  const isUserMentee = useMemo(() => role_id === 4, [role_id]);
+  /**
+   * Khaleel Musleh
+   * Created a Dev role_id which displays all the sidebar links of Admin, Mentor, Mentee
+   */
+  const isUserDev = useMemo(() => role_id === 5, [role_id]);
```
Comment (ryan-hamblin) [inline src/components/common/Sidebar/Sidebar.js:56]: is role_id===1 reserved only for superAdmin? Curious about 2-5 approach.

File: src/components/common/Sidebar/SidebarLinks.utils.js [hunk 2]
```diff
@@ -42,11 +42,20 @@ export const bottomSharedLinks = [
...
+ * Khaleel Musleh
+ * Uncommented all the links which are non-functional, they are all available in the devLinks below
+ */
+
 export const menteeLinks = [
-  {
-    key: '/assignments',
-    label: 'My Assignments',
-  },
+  // {
```
Comment (ryan-hamblin) [inline src/components/common/Sidebar/SidebarLinks.utils.js:51]: let's go ahead and remove this before merging.

File: src/components/common/Sidebar/SidebarLinks.utils.js [hunk 3]
```diff
@@ -58,10 +67,10 @@ export const menteeLinks = [
 ];
 
 export const mentorLinks = [
-  {
-    key: '/mentees',
-    label: 'My Mentees',
-  },
+  // {
```
Comment (ryan-hamblin) [inline src/components/common/Sidebar/SidebarLinks.utils.js:70]: please remove commented out code

File: src/components/common/Sidebar/SidebarLinks.utils.js [hunk 4]
```diff
@@ -70,6 +79,18 @@ export const mentorLinks = [
 ];
 
 export const adminLinks = [
+  // { key: '/dashboard', label: 'Dashboard', icon: <DashboardOutlined /> },
```
Comment (ryan-hamblin) [inline src/components/common/Sidebar/SidebarLinks.utils.js:82]: please remove commented out code.

File: src/components/pages/Dashboard/Dashboard.js [hunk 5]
```diff
@@ -1,127 +1,43 @@
...
+            ? 'Mentor'
+            : res.data.role_id === 4
+            ? 'Mentee'
+            : res.data.role_id === 5
+            ? 'Developer'
+            : 'pending'
+        );
+      })
+      .catch(err => {
+        console.log(err);
```
Comment (ryan-hamblin) [inline src/components/pages/Dashboard/Dashboard.js:32]: currently this error is simply being logged out to the console. lets return handle it for the user properly. 

You could `console.error` here for intents and purposes of shipping, however, I'd recommend showing the user that something went wrong. eventually that's what we want to do with our errors. why not now?

File: src/components/pages/TicketsDashboard/TicketsDashboard.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  {
+    title: 'Date Submitted',
+    dataIndex: 'dateSubmitted',
+    defaultSortOrder: 'descend',
+    sorter: (a, b) => a.dateSubmitted < b.dateSubmitted,
+  },
+];
+
+function onChange(pagination, filters, sorter, extra) {
+  // console.log('params', pagination, filters, sorter, extra);
```
Comment (ryan-hamblin) [inline src/components/pages/TicketsDashboard/TicketsDashboard.js:65]: do we need this func? if all it's doing is `console.logging`?

File: src/components/pages/TicketsDashboard/TicketsDashboard.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    };
+    getTickets();
+  }, []);
+
+  const data = [];
+  let escaTickets = [];
+  if (tickets !== []) {
+    escaTickets = tickets.filter(x => x.ticket_status === 'approved');
+  }
+  // console.log('escaTickets', escaTickets);
```
Comment (ryan-hamblin) [inline src/components/pages/TicketsDashboard/TicketsDashboard.js:90]: please remove

File: src/components/pages/TicketsDashboard/TicketsDashboard.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+  // eslint-disable-next-line array-callback-return
```
Comment (ryan-hamblin) [inline src/components/pages/TicketsDashboard/TicketsDashboard.js:91]: Please don't disable the linter. conform to it's standards :)

File: src/components/pages/TicketsDashboard/TicketsDashboard.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+  tickets.map(t => {
+    const ticketDetails = {
+      key: t.ticket_id,
+      ticketID: t.ticket_id,
+      message: t.ticket_subject,
+      // dateSubmitted: t.created_at.substring(0, 10),
```
Comment (ryan-hamblin) [inline src/components/pages/TicketsDashboard/TicketsDashboard.js:97]: please remove

**Review by ryan-hamblin — CHANGES_REQUESTED** [review]
Great work here adding back in the Rolechecks to hide/show content appropriately. 

There are a few instances of commented out code and console.logs that I want you to remove. There are also quite a few linter warnings (not related to work you've added) that I'd love addressed and some commented out code from previous commits in `src/components/common/Sidebar/Sidebar.js`

Please address those as this work was done in those files. Let's just clean things up a bit. 

I saw no deviations from industry practices that were concerning however, in `TicketDashboard.js` we're defining a function that doesn't seem to be used for anything more than `logging` the arguments. Let's remove that too.

File: src/components/pages/Dashboard/Dashboard.js (hunk 5 above)
Comment (ryan-hamblin) [reply src/components/pages/Dashboard/Dashboard.js:32]: no need to fix right away, but I'd love to know more about what you plan to do in the future to handle errors back to users.

File: src/components/common/Sidebar/Sidebar.js (hunk 1 above)
Comment (khaleel737) [reply src/components/common/Sidebar/Sidebar.js:53]: Hi Ryan, Ive made another strict equality check for super admin which is role_id === 1
they all now have a check from 1 - 5

File: src/components/common/Sidebar/SidebarLinks.utils.js (hunk 2 above)
Comment (khaleel737) [reply src/components/common/Sidebar/SidebarLinks.utils.js]: Removed it on last commit

File: src/components/common/Sidebar/SidebarLinks.utils.js (hunk 3 above)
Comment (khaleel737) [reply src/components/common/Sidebar/SidebarLinks.utils.js]: Removed it on last commit

File: src/components/common/Sidebar/SidebarLinks.utils.js (hunk 4 above)
Comment (khaleel737) [reply src/components/common/Sidebar/SidebarLinks.utils.js]: Removed it on last commit

File: src/components/pages/Dashboard/Dashboard.js (hunk 5 above)
Comment (khaleel737) [reply src/components/pages/Dashboard/Dashboard.js]: Ive changed it to console.error, its much better like this, thank you for noticing it

Comment (khaleel737) [reply src/components/pages/Dashboard/Dashboard.js]: Im now using state slice and dispatch to handle errors for API requests

File: src/components/pages/TicketsDashboard/TicketsDashboard.js (hunk 6 above)
Comment (khaleel737) [reply src/components/pages/TicketsDashboard/TicketsDashboard.js]: This component was dashboard before I changed it to ticketsdashboard, apologies but I didnt pay attention to the code as It was done by somebody else, I just changed Its directory and name, I removed all the console logs and functions that are not working there

File: src/components/pages/TicketsDashboard/TicketsDashboard.js (hunk 7 above)
Comment (khaleel737) [reply src/components/pages/TicketsDashboard/TicketsDashboard.js]: Removed it on last commit

File: src/components/pages/TicketsDashboard/TicketsDashboard.js (hunk 8 above)
Comment (khaleel737) [reply src/components/pages/TicketsDashboard/TicketsDashboard.js:86]: Fixed it on last commit

File: src/components/pages/TicketsDashboard/TicketsDashboard.js (hunk 9 above)
Comment (khaleel737) [reply src/components/pages/TicketsDashboard/TicketsDashboard.js]: Removed it on last commit

**Pull request merged**